File: aoj/volume100/ocaml/A10017.ml
let n_ways n x =
    let count = ref 0 in
    for a = 1 to n do
        for b = a+1 to n do
            for c = b+1 to n do
                if a + b + c == x then
                    count := !count + 1
            done ;
        done ;
    done ;
    !count

let solve n x =
    Printf.printf "%d\n" (n_ways n x)

let () =
    let rec loop () =
        let (n, x) = Scanf.scanf "%d %d\n" (fun x y -> (x, y)) in
        match (n, x) with
        | (0, 0) -> ()
        | (_, _) -> (
            solve n x ;
            loop () ;
        )
    in 
    loop ()

File: aoj/volume100/cpp/A10017.cpp
#include <iostream>

auto main() -> int {
	int n, x;
	for (;;) {
		std::cin >> n >> x;
		if (n == 0 && x == 0) {
			break;
		}

		int count = 0;
		for (int a = 1;   a <= n; ++a)
		for (int b = a+1; b <= n; ++b)
		for (int c = b+1; c <= n; ++c)
			if (a + b + c == x)
				count++;

		std::cout << count << std::endl;
	}
	return 0;
}
